pkg/querier/querier.go

```diff
@@ -97,8 +99,21 @@ func New(cfg Config, distributor Distributor, chunkStore ChunkStore) (storage.Qu
 }
 
 // NewQueryable creates a new Queryable for cortex.
-func NewQueryable(dq, cq storage.Queryable, distributor Distributor, ingesterMaxQueryLookback time.Duration) storage.Queryable {
+func NewQueryable(dq, cq storage.Queryable, distributor Distributor, ingesterMaxQueryLookback, maxQueryIntoFuture time.Duration) storage.Queryable {
 	return storage.QueryableFunc(func(ctx context.Context, mint, maxt int64) (storage.Querier, error) {
+		now := time.Now()
+
+		if maxQueryIntoFuture > 0 {
+			maxQueryTime := util.TimeMilliseconds(now.Add(maxQueryIntoFuture))
+
+			if mint > maxQueryTime {
+				return storage.NoopQuerier(), nil
+			}
+			if maxt > maxQueryTime {
+				maxt = maxQueryTime
+			}
+		}
+
```

Comment:
is there somewhere in the querier package for this to live so it doesn't have to be duplicated in `newUnifiedChunkQueryable`?